pennylane/ops/qubit/parametric_ops.py

```diff
@@ -525,16 +525,12 @@ def eigvals(self):
         return self._eigvals(*self.parameters, len(self.wires))
 
     @staticmethod
-    def decomposition(theta, wires):
-        with qml.tape.OperationRecorder() as rec:
-            for i in range(len(wires) - 1, 0, -1):
-                qml.CNOT(wires=[wires[i], wires[i - 1]])
-
-            RZ(theta, wires=wires[0])
+    def compute_decomposition(theta, wires):
+        ops = [qml.CNOT(wires=(w0, w1)) for w0, w1 in zip(wires[~0:0:-1], wires[~1::-1])]
+        ops.append(RZ(theta, wires=wires[0]))
+        ops += [qml.CNOT(wires=(w0, w1)) for w0, w1 in zip(wires[1:], wires[:~0])]
```

Comment:
:100:

Reply:
nice!

Reply:
re-reading this though, I find it a bit harder to read/parse compared to the left 🤔 

How come you use `~0` instead of -1, and `~1` instead of -2?

Reply:
I learned about using `~` recently and personally prefer it.  It's more symmetric. `a[~0]` is just `a[0]` read from the end. But again, this can just use `-1` and `-2` since that's what is more common.